aspnet/migration/http-modules.rst

```diff
@@ -1,8 +1,584 @@
+	:language: json
+	:linenos:
+	:lines: 1, 6-10
+
+*MyMiddlewareOptionsSection* here is simply a section name. It doesn't have to be the same as the name of your options class.
+
+3. Associate the option values with the options class
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The options pattern uses ASP.NET 5's dependency injection framework to associate the options type (such as ``MyMiddlewareOptions``)
+with an ``MyMiddlewareOptions`` object that has the actual options.
+
+To make this work, update your ``Startup`` class:
+
+	a. If you're using *appsettings.json*, make sure it is added to the configuration builder in the ``Startup`` constructor:
+
+	.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Startup.cs
+		:language: c#
+		:linenos:
+		:lines: 14-23, 106, 109
+		:emphasize-lines: 7
+		:dedent: 4
+
+	b. Make sure that the options service is configured:
+
+	.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Startup.cs
+		:language: c#
+		:linenos:
+		:lines: 14-15, 28-29, 31-33, 43, 109
+		:emphasize-lines: 5
+		:dedent: 4
+
+	c. Associate your options with your options class:
+
+	.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Startup.cs
+		:language: c#
+		:linenos:
+		:lines: 14-15, 28-29, 31-32, 36-39, 43, 109
+		:emphasize-lines: 7,8
+		:dedent: 4
+
+4. Inject the options into your middleware constructor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Finally inject the options into your middleware's constructor. This is similar to injecting options into a controller.
+
+.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/MyMiddlewareWithParams.cs
+	:language: c#
+	:linenos:
+	:lines: 6-7, 24-47
+	:emphasize-lines: 7,10,13,19,24
+
+This works, because the `UseMiddleware <#http-modules-usemiddleware>`_ extension method that adds your middleware to the
+``IApplicationBuilder`` takes care of dependency injection.
+
+This is not limited to ``IOptions`` objects. Any other object that your middleware requires can be dependency injected this way.
+
+Loading middleware options through direct injection
+---------------------------------------------------
+The options pattern has the advantage that it creates loose coupling between options values and their consumers.
+Once you've associated an options class with the actual options values, any other class can get access to the options
+through the dependency injection framework. So there is no need to pass around options values.
+
+This breaks down though if you want to use the same middleware twice, with different options. For example an authorization middleware
+used in different branches allowing different roles. You can't associate two different options objects with the one options class.
+
+The solution is to get the options objects with the actual options values in your ``Startup`` class and pass those directly to each instance of your middleware.
+
+1. Add a second key to appsettings.json
```

Comment:
You have two options: 1. Remove the section heading formatting (the `^^^` underline) and indent everything under the numbered bullet so that it is rendered as nested content, or 2. Remove the numbering form the section headings. I tried out option (1) and I think it looks fine (see https://github.com/aspnet/Docs/tree/danroth27/http-modules)